source/material_model/utilities.cc

```diff
@@ -514,58 +564,39 @@ namespace aspect
           enthalpy_values.reinit(n_temperature,n_pressure);
 
           unsigned int i = 0;
+          std::vector<double> previous_row_values(n_columns, 0.);
+
           while (!in.eof())
             {
-              double temp1,temp2;
-              double rho,alpha,cp,vp,vs,h;
-              in >> temp1 >> temp2;
-              in >> rho;
-              if (in.fail())
-                {
-                  in.clear();
-                  rho = density_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> alpha;
-              if (in.fail())
-                {
-                  in.clear();
-                  alpha = thermal_expansivity_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> cp;
-              if (in.fail())
-                {
-                  in.clear();
-                  cp = specific_heat_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> vp;
-              if (in.fail())
-                {
-                  in.clear();
-                  vp = vp_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> vs;
-              if (in.fail())
-                {
-                  in.clear();
-                  vs = vs_values[(i-1)%n_temperature][(i-1)/n_temperature];
-                }
-              in >> h;
-              if (in.fail())
+              std::vector<double> row_values(n_columns);
+
+              for (unsigned int n=0; n<n_columns; n++)
                 {
-                  in.clear();
-                  h = enthalpy_values[(i-1)%n_temperature][(i-1)/n_temperature];
+                  in >> row_values[n];
+                  // PerpleX P-T grids sometimes contain rows filled with NaNs
+                  // in the corners of the grid.
+                  // These P-T regions are unimportant, and we don't want ASPECT
+                  // to crash just because it sees a NaN.
+                  // Here we choose to replace invalid doubles
+                  // with the most recent valid double.
+                  if (in.fail())
+                    {
+                      in.clear();
+                      row_values[n] = previous_row_values[n];
```

Comment:
You're basically padding out the values with valid values. Wouldn't it make more sense to also set these values as `nan` and hope that we never actually query them in a simulation? (Because if we ever did, we can't claim that the values we return with this scheme are actually "correct".)

Reply:
I'm a bit concerned that this might break existing prms. In addition to the corner issue, PerpleX has a habit of producing single P-T pixels of NaN (for a reason I genuinely don't understand). In these cases, selecting the previous row is a reasonable approach. Of course, it would be better to preprocess the tables so that they don't contain these dodgy NaNs, but at this point, the PerpleX lookup has been in ASPECT since 2015...

For now, I'll extend this comment to include the above, but if you'd like me to change the functionality, then I can.

Reply:
No, that sounds reasonable if you think you have good reasons. Let's go with what you have then.

Is this documented in the description of the plugin?